### ISSUE TITLE
Contributing to this repository

### ISSUE THREAD

**Issue by kabirnagral** [cashmusic/platform#1186]
Could please tell me the latest instructions for developing and testing a branch? There's two methods to contribute and test the code locally that conflict and I'm not sure which one to use.

**Comment by paperscissors** [comment]
Sure thing --- sorry everything is in such disarray --- we've been pretty short staffed and I've been focusing on modernizing a lot of the legacy code.

I'm pretty sure when I added modern namespacing it broke the docs, but ultimately there are some big changes to consider that I'm in the process of folding in from staging. Unit tests are funky at the moment for the same reason.

In terms of actually running the platform locally, we have a [Vagrant box here](https://github.com/cashmusic/rackspace-template) that has known issues on Fedora and Ubuntu; there are some issues with some rubygems during puppet deployment that need to be fixed manually at this point. It's a drag honestly. The instructions for setup are pretty accurate, though. I use a raspberry pi LAMP setup here for testing, which is also not ideal.

My goal after I deploy this big modernization branch is to put together a Docker environment that is more reliable.

The other caveat is you probably want to branch off of staging because I've got some significant ORM changes that deviate (in a good way) from the docs.

Long story short, I can get the docs up and running quickly, but there are some other parts I might need to bring you up to speed on. Things are in a transition right now, so the takeaway of this is pardon our mess.
